### PR TITLE
Rewrite dev guide

### PR DESCRIPTION
**Pull request by armincerf**

### CODE REVIEW

File: doc/resources/doc/sources/dev-guide.adoc [hunk 1]
```diff
@@ -1,144 +1,203 @@
...
 ----
-`reset`:: Use this to start the system, and after making code changes.  It will reload your code, and your system.
-+
-[source,clojure-repl]
+
+== Making code changes
+
+When developing Clojure will reload the code inside of the running process, instead of restarting the process.
+To do this, type `(reset)` at the REPL.
+
```
Comment (armincerf) [inline doc/resources/doc/sources/dev-guide.adoc]: Worth mentioning you can add `(cider-interactive-eval "(dev-extras/reset)")` to a keybinding to make this process easier in emacs

File: doc/resources/doc/sources/dev-guide.adoc [hunk 2]
```diff
@@ -1,144 +1,203 @@
...
 
-== REPLant.nvim
+This will result in 2 REPL buffers, one for ClojureScript, and one for Clojure.
+To setup your Clojure REPL, follow the <<editor_clj_repl,Editor REPL instructions>>.
+
+[[nrepl_clojurescript]]
+=== nREPL
 
-REPLant's refresh is fully compatible with this setup and will work automatically.
+You can
```
Comment (armincerf) [inline doc/resources/doc/sources/dev-guide.adoc]: ...

File: doc/resources/doc/sources/dev-guide.adoc [hunk 3]
```diff
@@ -1,144 +1,203 @@
...
-(ns dev
-  (:require
-    [dev-extras :refer :all]))
+user=>
 ----
-. Create a `.dir-locals.el` containing your dev jack-in aliases
-+
-[source,elisp]
+
+Here, you should type `(dev)` to get to the dev namespace:
```
[the remark this answers is not in the document]
Comment (SevereOverfl0w) [reply doc/resources/doc/sources/dev-guide.adoc]: Going to add this further down in the "starting the system" section.

**Comment by SevereOverfl0w** [comment]
I think this is good for re-review.

**Pull request merged**